lib/chef/provider/ifconfig.rb

```diff
@@ -111,7 +111,7 @@ def load_current_resource
           #       TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
           @status = shell_out("ifconfig")
           @status.stdout.each_line do |line|
-            addr_regex = /^(\w+):?(\d*):?\ .+$/
+            addr_regex = /^(\w+)(-?):?(\d*):?\ .+$/
```

Comment:
Can you attack a permalink from https://rubular.com/ that shows the various interface names we're trying to match so it's more clear to those in the future what this regex does?

Reply:
@tas50 Here is the list of interfaces that we are trying to match. Also added output of `ifconfig` so it's more clear.

Output of `ifconfig` after creating `Bridge interface` i.e. **br-1234**
```
[root@ip-172-31-26-90 chef]# ifconfig
br-1234: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        inet 52.34.83.20  netmask 255.255.255.255  broadcast 52.34.83.20
        ether 02:6a:d0:5d:0d:bd  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 9001
        inet 172.31.26.90  netmask 255.255.240.0  broadcast 172.31.31.255
        inet6 fe80::2e:4eff:fe78:6c90  prefixlen 64  scopeid 0x20<link>
        ether 02:2e:4e:78:6c:90  txqueuelen 1000  (Ethernet)
        RX packets 454657  bytes 604056536 (576.0 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 87340  bytes 13338865 (12.7 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

**Before fix permalink** : https://rubular.com/r/ujRptRHoxft64A
**After fix permalink** :  https://rubular.com/r/l9C8YlVgHO4cHN

Reply:
Wouldn't it be more appropriate to use a match like https://rubular.com/r/JrykUpfjRnYeQD that would actually match on the full interface name.  Also can you  put  the actual permalink directly in the code comments so future developers can see what we're trying to match.

Reply:
Thank you @tas50, Updated!